### ISSUE TITLE
Reproducing LoRRA results on TextVQA

### ISSUE THREAD

**Issue by natsudalkr** [facebookresearch/pythia#85]
I am trying to reproduce the results using this config file. https://github.com/facebookresearch/pythia/blob/master/configs/vqa/textvqa/lorra.yml
The only thing I changed is num_worker because I don't have enough memory (I changed num_worker to 2) but the accuracy I got is not as high as the pre-trained one. I only got around 0.16-0.17 on validation data while the pre-trained model got ~0.27.
Do you have any suggestions or ideas about what I have done wrong or what I have missed?

Thank you very much!

**Comment by apsdehal** [comment]
@natsudalkr Can you post the exact command you are using here as well as your config file?

**Comment by natsudalkr** [comment]
Sorry for the late response.

The command is 
```python tools/run.py --tasks vqa --datasets textvqa --model lorra --config configs/vqa/textvqa/lorra.yml```

Below is the config file I used. I have tried with a few learning rate settings but got quite similar results. 
Thank you very much!
```
includes:
- common/defaults/configs/tasks/vqa/textvqa.yml
# Use soft copy
task_attributes:
  vqa:
    dataset_attributes:
      textvqa:
        use_ocr: true
        processors:
          answer_processor:
            type: soft_copy_answer
            params:
              vocab_file: vocabs/answers_textvqa_more_than_1.txt
              preprocessor:
                type: simple_word
                params: {}
              context_preprocessor:
                type: simple_word
                params: {}
              max_length: 50
              num_answers: 10
model_attributes:
  lorra: &lorra
    model_data_dir: ../data
    metrics:
    - type: vqa_accuracy
    losses:
    - type: logit_bce
    num_context_features: 1
    context_feature_dim: 300
    image_feature_dim: 2048
    context_max_len: 50
    classifier:
      type: logit
      params:
        img_hidden_dim: 5000
        text_hidden_dim: 300
    image_feature_embeddings:
    - modal_combine:
        type: non_linear_element_multiply
        params:
          dropout: 0
          hidden_dim: 5000
      normalization: softmax
      transform:
        type: linear
        params:
          out_dim: 1
    context_feature_embeddings:
    - modal_combine:
        type: non_linear_element_multiply
        params:
          dropout: 0
          hidden_dim: 5000
      normalization: sigmoid
      transform:
        type: linear
        params:
          out_dim: 1
    image_feature_encodings:
    - type: finetune_faster_rcnn_fpn_fc7
      params:
        bias_file: detectron/fc6/fc7_b.pkl
        weights_file: detectron/fc6/fc7_w.pkl
    - type: default
      params: {}
    context_feature_encodings:
    - type: default
      params: {}
    image_text_modal_combine:
      type: non_linear_element_multiply
      params:
        dropout: 0
        hidden_dim: 5000
        # 300 for FastText and 50 for order vectors
        context_dim: 350
    text_embeddings:
    - type: attention
      params:
        hidden_dim: 1024
        num_layers: 1
        conv1_out: 512
        conv2_out: 2
        dropout: 0
        embedding_dim: 300
        kernel_size: 1
        padding: 0
    context_embeddings:
    - type: identity
      params:
        embedding_dim: 350
  lorra_with_glove: *lorra
optimizer_attributes:
  params:
    eps: 1.0e-08
    lr: 0.05
    weight_decay: 0
  type: Adamax
training_parameters:
    clip_norm_mode: all
    clip_gradients: true
    max_grad_l2_norm: 0.25
    lr_scheduler: true
    lr_steps:
    - 14000
    lr_ratio: 0.01
    use_warmup: true
    warmup_factor: 0.2
    warmup_iterations: 1000
    max_iterations: 24000
    batch_size: 128
    num_workers: 2
    task_size_proportional_sampling: true
    monitored_metric: vqa_accuracy
    pretrained_mapping:
      text_embeddings: text_embeddings
      image_feature_encoders: image_feature_encoders
      image_feature_embeddings_list: image_feature_embeddings_list
      image_text_multi_modal_combine_layer: image_text_multi_modal_combine_layer
    load_pretrain: false
    metric_minimize: false
    data_parallel: true
    evalai_inference: true
```

**Comment by fliu2013** [comment]
I found that the same code in pytorch 0.4 and pytorch 1.0 performs differently. I run `python tools/run.py --tasks vqa --datasets textvqa --model lorra --config configs/vqa/textvqa/lorra.yml` in pytorch 0.4 and got around 0.16 . Pytorch 1.0 gives better results. Maybe you can give a try.

**Comment by natsudalkr** [comment]
@fliu2013 Thank you very much for your advice. 
I used pytorch 1.0.0 for this project (and got ~0.16-0.17).

**Comment by apsdehal** [comment]
Yes, this is a known issue. I am unable to reproduce the bug in my setup environment.  @natsudalkr can you run the command below and paste the output here:

```
# For security purposes, please check the contents of collect_env.py before running it.
python -m torch.utils.collect_env
```

**Comment by natsudalkr** [comment]
@apsdehal Here is the output.
```
Collecting environment information...
PyTorch version: 1.0.0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
CMake version: Could not collect

Python version: 3.5
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1080 Ti
GPU 2: GeForce GTX 1080 Ti
GPU 3: GeForce GTX 1080 Ti

Nvidia driver version: 410.104
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.5.0
/usr/lib/x86_64-linux-gnu/libcudnn_static_v7.a

Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect
```

**Comment by apsdehal** [comment]
@natsudalkr I tried reproducing your bug in my environment but couldn't. Can you try following things to check whether it helps:
- Your CUDA used to build PyTorch and your runtime CUDA version are different so reinstall pytorch to make them same.
- Use torch==1.0.1.post2 version by running `pip install torch==1.0.1.post2`.

Let me know if it helps.

**Comment by apsdehal** [comment]
@natsudalkr Attaching the log file for reference here.
[vqa_textvqa_lorra_1234_2019-06-11T09:56:03.log](https://github.com/facebookresearch/pythia/files/3282509/vqa_textvqa_lorra_1234_2019-06-11T09.56.03.log)

**Comment by natsudalkr** [comment]
By changing pytorch version to torch==1.0.1.post2, now it works! 
(I haven't fixed the CUDA version mismatch yet)

Thank you very much. @apsdehal @fliu2013

**Comment by apsdehal** [comment]
Great. Closing then. Open up a new issue if you face any other problems.
